contracts/interfaces/IQuoter.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+import "../libraries/SwapIntention.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+
+interface IQuoter {
+    error InvalidQuoteType();
+    error InvalidQuoteTypeInRevert();
+    error UnexpectedRevertBytes();
+
+    struct NonZeroDeltaCurrency {
+        Currency currency;
+        int128 deltaAmount;
+    }
+
+    function quoteExactInputSingle(ExactInputSingleParams calldata params)
```

Comment:
nit: natspec here, then you can reference it from the implementation with @inheritdoc